src/Ampersand/PatchHelper/Patchfile/Entry.php

```diff
@@ -263,4 +263,19 @@ public function __toString()
     {
         return implode(PHP_EOL, $this->lines);
     }
+
+    public function applyToTheme($projectDir, $error, $fuzzFactor)
+    {
+        $tmpPatchFilePath = rtrim($projectDir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . 'tmp.patch';
+        $adaptedLines     = [];
+        foreach ($this->lines as $line) {
+            // Replace files in lines to actually apply the changes to the current theme
+            $adaptedLine     = str_replace($this->newFilePath, str_replace($projectDir, '', $error), $line);
+            $adaptedLines [] = str_replace('vendor_orig', 'vendor', $adaptedLine);
+        }
+        file_put_contents($tmpPatchFilePath, implode(PHP_EOL, $adaptedLines));
+        shell_exec('patch < ' . $tmpPatchFilePath . ' -p0 -F' . $fuzzFactor . ' --no-backup-if-mismatch -d'
```

Comment:
So far the only other usage of `shell_exec` is executing `find`.

I've got a sanity check that `find` is installed at this location
https://github.com/AmpersandHQ/ampersand-magento2-upgrade-patch-helper/blob/master/src/functions.php#L3-L5

Would be good to have a similar sanity check for `patch`

Reply:
@convenient , thanks for your feedback.
Added a check